### DISCUSSION TITLE
Regarding array type input entry in decision table

### DISCUSSION THREAD

**Discussion by aayushbhaglal** [gorules/zen#195] [Q&A] [answered]
Hello everyone,

I am having trouble in using array type input in my decision table (I am using an array of 7000 pincodes). However, whenever I try to use smaller arrays of size 5-10 it works flawlessly but whenever I try to use the complete dataset of 10000-20000 it starts throwing the following error : 
SIGSEGV: segmentation violation
PC=0x1024bcdbc m=0 sigcode=2
signal arrived during cgo execution

goroutine 1 [syscall]:
runtime.cgocall(0x102380a88, 0x140001d1d68)
        /usr/local/go/src/runtime/cgocall.go:157 +0x44 fp=0x140001d1d30 sp=0x140001d1cf0 pc=0x1022b91e4
github.com/gorules/zen-go._Cfunc_zen_decision_evaluate(0x112706c60, 0x112704d40, {0x0, 0x1})
        _cgo_gotypes.go:160 +0x3c fp=0x140001d1d60 sp=0x140001d1d30 pc=0x10237f42c
github.com/gorules/zen-go.decision.EvaluateWithOpts({0x0?}, {0x10268e060?, 0x14000184120?}, {0x20?, 0xe1?})
        /Users/aayush/go/pkg/mod/github.com/gorules/zen-go@v0.3.0/decision.go:41 +0xc4 fp=0x140001d1e70 sp=0x140001d1d60 pc=0x10237f724
github.com/gorules/zen-go.decision.Evaluate({0x10268e060?}, {0x10268e060?, 0x14000184120?})
        /Users/aayush/go/pkg/mod/github.com/gorules/zen-go@v0.3.0/decision.go:24 +0x2c fp=0x140001d1ea0 sp=0x140001d1e70 pc=0x10237f62c
main.main()
        /Users/aayush/Desktop/main.go:30 +0x214 fp=0x140001d1f30 sp=0x140001d1ea0 pc=0x1023808a4
runtime.main()
        /usr/local/go/src/runtime/proc.go:267 +0x2bc fp=0x140001d1fd0 sp=0x140001d1f30 pc=0x1022e973c
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x140001d1fd0 sp=0x140001d1fd0 pc=0x102315b34

goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:398 +0xc8 fp=0x1400004af90 sp=0x1400004af70 pc=0x1022e9b68
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:404
runtime.forcegchelper()
        /usr/local/go/src/runtime/proc.go:322 +0xb8 fp=0x1400004afd0 sp=0x1400004af90 pc=0x1022e99f8
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x1400004afd0 sp=0x1400004afd0 pc=0x102315b34
created by runtime.init.6 in goroutine 1
        /usr/local/go/src/runtime/proc.go:310 +0x24

goroutine 3 [GC sweep wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:398 +0xc8 fp=0x1400004b760 sp=0x1400004b740 pc=0x1022e9b68
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:404
runtime.bgsweep(0x0?)
        /usr/local/go/src/runtime/mgcsweep.go:280 +0xa0 fp=0x1400004b7b0 sp=0x1400004b760 pc=0x1022d6e00
runtime.gcenable.func1()
        /usr/local/go/src/runtime/mgc.go:200 +0x28 fp=0x1400004b7d0 sp=0x1400004b7b0 pc=0x1022cbb88
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x1400004b7d0 sp=0x1400004b7d0 pc=0x102315b34
created by runtime.gcenable in goroutine 1
        /usr/local/go/src/runtime/mgc.go:200 +0x6c

goroutine 4 [GC scavenge wait]:
runtime.gopark(0x140000240e0?, 0x1025afde8?, 0x1?, 0x0?, 0x140000031e0?)
        /usr/local/go/src/runtime/proc.go:398 +0xc8 fp=0x1400004bf50 sp=0x1400004bf30 pc=0x1022e9b68
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:404
runtime.(*scavengerState).park(0x10273e180)
        /usr/local/go/src/runtime/mgcscavenge.go:425 +0x5c fp=0x1400004bf80 sp=0x1400004bf50 pc=0x1022d46ac
runtime.bgscavenge(0x0?)
        /usr/local/go/src/runtime/mgcscavenge.go:653 +0x44 fp=0x1400004bfb0 sp=0x1400004bf80 pc=0x1022d4c04
runtime.gcenable.func2()
        /usr/local/go/src/runtime/mgc.go:201 +0x28 fp=0x1400004bfd0 sp=0x1400004bfb0 pc=0x1022cbb28
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x1400004bfd0 sp=0x1400004bfd0 pc=0x102315b34
created by runtime.gcenable in goroutine 1
        /usr/local/go/src/runtime/mgc.go:201 +0xac

goroutine 18 [finalizer wait]:
runtime.gopark(0x29?, 0x0?, 0xe8?, 0xa5?, 0x10233a7dc?)
        /usr/local/go/src/runtime/proc.go:398 +0xc8 fp=0x1400004a580 sp=0x1400004a560 pc=0x1022e9b68
runtime.runfinq()
        /usr/local/go/src/runtime/mfinal.go:193 +0x108 fp=0x1400004a7d0 sp=0x1400004a580 pc=0x1022cac38
runtime.goexit()
        /usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x1400004a7d0 sp=0x1400004a7d0 pc=0x102315b34
created by runtime.createfing in goroutine 1
        /usr/local/go/src/runtime/mfinal.go:163 +0x80

r0      0x16d350c20
r1      0x16db49e60
r2      0x11811aaa0
r3      0x7f818
r4      0x4
r5      0x20
r6      0x0
r7      0x16db4b008
r8      0x1
r9      0x1025b755a
r10     0x1024bd120
r11     0xc7
r12     0x1181cffc0
r13     0x102843200
r14     0x2
r15     0x1
r16     0x181c1187c
r17     0x1f0870270
r18     0x0
r19     0x16d351400
r20     0x16db49e60
r21     0x1182b07f8
r22     0x8000000000000002
r23     0x3
r24     0x16d350570
r25     0x16d350c20
r26     0x11811a9f8
r27     0x1
r28     0x1
r29     0x16d3504d0
lr      0x1024bd4f4
sp      0x16d34fd00
pc      0x1024bcdbc
fault   0x16d34fd00
exit status 2

I am attaching the input json for your reference
[graph10.json](https://github.com/user-attachments/files/16029224/graph10.json)

Thanks

**Comment by ivanmiletic** [comment]
We will take a look

**Comment by ivanmiletic** [comment, marked as answer]
Hi @aayushbhaglal 

This is a known issue, for rules of this size use arrays directly, for example instead of using:

```
1,2,3,4
```

use

```
[1,2,3,4]
```

In your example just add square brackets at the beginning and end.

**Reply by aayushbhaglal** [reply]
It worked. Thanks !!!
